File: go/design-patterns/composite/composite/directory_composite.go
type DirectoryComposite struct {
    name     string
    children []FileSystemNode
}
func (d *DirectoryComposite) GetName() string { return d.name }
func (d *DirectoryComposite) Add(node FileSystemNode) {
    d.children = append(d.children, node)
}
func (d *DirectoryComposite) GetSize() int {
    sum := 0
    for _, node := range d.children {
        sum += node.GetSize()
    }
    return sum
}

File: java/design-patterns/composite/composite/DirectoryComposite.java
class DirectoryComposite implements FileSystemNode {
    private String name;
    private List<FileSystemNode> children = new ArrayList<>();
    public DirectoryComposite(String name) { this.name = name; }
    public String getName() { return name; }
    public void add(FileSystemNode node) { children.add(node); }
    public int getSize() {
        int sum = 0;
        for (FileSystemNode node : children) {
            sum += node.getSize();
        }
        return sum;
    }
}